src/services/image.service.ts

```diff
@@ -42,8 +42,33 @@ export class ImageService {
     }).save();
   }
 
-  public async readRecipeImages(id: string) {
-    return Image.find({ recipeId: id });
+  public async readRecipeImages(recipeId: string) {
+    return Image.find({ recipeId });
+  }
+  public async readRecipeImage(imageId: string): Promise<Buffer | null> {
+    const image = await Image.findById(imageId);
+    if (!image) {
+      return null;
+    }
+
+    return image.image;
+  }
+
+  public async getUrls(recipeId: string): Promise<string[] | null> {
+    let urls: Array<string> = [];
+
+    const images = await this.readRecipeImages(recipeId);
+
+    if (images?.length === 0) {
+      return null;
+    }
+
+    images.forEach((element) => {
+      urls.push(
+        `https://mooduprecipeapi.herokuapp.com/image/${element._id}/display`
```

Comment:
The URL should not be hardcoded. When we change the hosting (e.g. from Heroku to AWS) this won't work anymore.

The best option would be to include the base URL to API in environmental variables, e.g.

```
API_BASE_URL=https://mooduprecipeapi.herokuapp.com
```

and construct the URL with it's value

```
urls.push(`${process.env.API_BASE_URL}/image/${element._id}/display`);
```

Reply:
You can start by moving base URL to const and then move it to .env file